compiler-core/src/diagnostic.rs

```diff
@@ -19,18 +21,37 @@ pub struct Label {
     pub span: SrcSpan,
 }
 
+trait ToCodespanLabel {
```

Comment:
No new traits please.

Reply:
oh good call I don't need this anymore. In an old version of this code I had this implemented on the locations